File: 349. Intersection of Two Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> result;
        unordered_set<int> hashMap(nums1.begin(), nums1.end());
        for(auto val : nums2) {
            if(hashMap.count(val)) {
                result.push_back(val);
                hashMap.erase(val);
            }
        }
        return result;
    }
};

File: 349. Intersection of Two Arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set = new HashSet<>();
        Set<Integer> intersect = new HashSet<>();

        for(int val : nums1) {
            set.add(val);
        }

        for(int val : nums2) {
            if(set.contains(val))intersect.add(val);
        }

        int[] result = new int[intersect.size()];

        int i = 0;
        for(int val : intersect) {
            result[i ++] = val;
        }

        return result;
    }
}
